### PR TITLE
Seeding rework

### PR DESCRIPTION
**Pull request by diversifiedmechanic**
It is now modular and returns an array of the requested amount of restaurants.

### CODE REVIEW

File: database/seed.js [hunk 1]
```diff
@@ -13,59 +11,60 @@ const circleColors = ['#6c8ae4', '#d86441', '#bb6acd', '#df4e96'];
...
+  review.ambience = faker.random.number({min:1, max: 5});
+  review.dineDate = moment(faker.date.between('2019-01-01', '2020-01-24')).format('YYYY-MM-DD');
+  review.noise = faker.random.number({min:1, max: 5});
+  review.recommend = Math.random() < 0.7;
+  review.comments = jabber.createParagraph(faker.random.number({min:10, max: 50}));
+  review.filterTag = faker.random.arrayElement(tags);
+
+  // person
+  review.numReviews = faker.random.number({min:1, max:25});
+  review.color = faker.random.arrayElement(circleColors);
```
Comment (naszzz) [inline database/seed.js]: will all this eventually be moved to a generate data file, and invoked in seed.js ?

File: database/seed.js [hunk 2]
```diff
@@ -13,59 +11,60 @@ const circleColors = ['#6c8ae4', '#d86441', '#bb6acd', '#df4e96'];
...
+
+  for (let i = 1; i <= restaurantCount; i += 1) {
+    let restaurant = {};
+    restaurant.city = faker.address.city();
+    restaurant.restaurantId = lastRestId + i;
+    restaurant.reviews = [];
+
+    console.log(restaurant.restaurantId);
+    const reviewCount = faker.random.number({min:5, max: 10});
+
```
Comment (naszzz) [inline database/seed.js]: lgtm.

File: database/dataGeneration.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
+const faker = require('faker/locale/en_US');
+const Jabber = require('jabber');
+const moment = require('moment');
+const fs = require('fs');
+
+const buzzWords = ['great food', 'eat', 'service', 'drinks', 'loved', 'never again', 'dessert', 'main menu', 'appetizers', 'enjoyed', 'hated', 'server', 'noise', 'waitress', 'waiter', 'slow', 'fast', 'go again', 'clean', 'dirty', 'kids', 'our group', 'slow service', 'cold food', 'awesome drinks', 'fun time', 'romantic atmosphere', 'dinner', 'lunch', 'breakfast', 'hot plate', 'smelled wonderful', 'no alcohol', 'bartender', 'wine list', 'outdoor seating', 'large room', 'no tips', 'valet parking', 'fine dining', 'dishes', 'shrimp', 'steak', 'chicken', 'beef', 'pasta', 'cake', 'pastries', 'creme brulee', 'simmering', 'coffee', 'fresh food', 'candlelit', 'anniversary', 'birthday', 'party', 'with friends', 'evening', 'stuffed', 'would go again', 'recommend this place', 'restrooms clean', 'host greeting', 'lovely place', 'best view', 'indoor seating', 'waited an hour', 'risotto', 'baked salmon', 'fried rice', 'shrimp scampi', 'sushi', 'wait time'];
+
+const tags = ['Good for groups', 'Desserts', 'Appetizers', 'Drinks', 'Kid friendly'];
+
```
Comment (ZacharyAb) [inline database/dataGeneration.js]: This is such a cool way to add some realism!

File: database/dataGeneration.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+const circleColors = ['#6c8ae4', '#d86441', '#bb6acd', '#df4e96'];
+
+const jabber = new Jabber(buzzWords, 2);
+
+const allWritesFinished = [false, false];
+
+
+
+async function createReviewersAndReviews(NumberOfPeople) {
+  let reviewerCount = NumberOfPeople || 100;
```
Comment (ZacharyAb) [inline database/dataGeneration.js]: I guess I don't see the point of having the "or" here.

File: database/dataGeneration.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        reviewer.lastName = (Math.random() < 0.7) ? faker.name.lastName() : '';
+      } else {
+        reviewer.lastName = '';
+      }
+      reviewerCount -= 1; // maybe move to below the review section?
+
+      ///////////////////////////////////////////
+
+      reviewIndex += await writeReview(reviewer.id, reviewer.reviewCount, writeReviews, reviewIndex);
+
```
Comment (ZacharyAb) [inline database/dataGeneration.js]: This is an interesting use of await, I can't wait to see what happens with the massive amount of data you're generating.

File: database/dataGeneration.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        );
+      }
+    } while (reviewerCount >= 0 && ok);
+
+    if (reviewerCount > 0) {
+      writeReviewers.once('drain', writeReviewerToCSV);
+    }
+  }
+};
+
```
Comment (ZacharyAb) [inline database/dataGeneration.js]: lgtm

File: database/dataGeneration.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+      review.food = faker.random.number({min:1, max: 5});
+      review.service = faker.random.number({min:1, max: 5});
+      review.ambience = faker.random.number({min:1, max: 5});
+      review.dineDate = moment(faker.date.between('2019-01-01', '2020-01-24')).format('YYYY-MM-DD');
+      review.noise = faker.random.number({min:1, max: 5});
+      review.recommend = Math.random() < 0.7;
+      review.comments = jabber.createParagraph(faker.random.number({min:10, max: 50}));
+      review.filterTag = faker.random.arrayElement(tags);
+      reviewIndex += 1;
+
```
Comment (ZacharyAb) [inline database/dataGeneration.js]: A personal preference, but this is hard to read. Maybe taking some of your values and setting them to variables would help, but overall the logic looks great.

File: database/dataGeneration.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    } while (restaurantCount >= 0 && ok);
+
+    if (restaurantCount > 0) {
+      writeFile.once('drain', writeToCSV);
+    }
+  }
+};
+
+createRestaurants(10000000);
```
Comment (ZacharyAb) [inline database/dataGeneration.js]: lgtm

File: database/mongo.js [hunk 9]
```diff
@@ -57,8 +55,6 @@ const addOneReview = (newReviewData) => {
         vip: newReviewData.vip,
         color: newReviewData.color,
       });
-
-
       return newReview.save(newReviewData);
     });
 };
```
Comment (ZacharyAb) [inline database/mongo.js]: lgtm

File: database/dataGeneration.js (hunk 4 above)
Comment (diversifiedmechanic) [reply database/dataGeneration.js]: It's legacy.  I dont touch things that aren't broken.

File: database/dataGeneration.js (hunk 5 above)
Comment (diversifiedmechanic) [reply database/dataGeneration.js]: I can tell you now, it won't end well.

File: database/dataGeneration.js (hunk 7 above)
Comment (diversifiedmechanic) [reply database/dataGeneration.js]: Alright, I'll look into it!

**Pull request merged**